.github/workflows/test-runner-pr-integration.yml

```diff
@@ -0,0 +1,103 @@
+# This workflow will build a Java project with Gradle
+# For more information see: https://help.github.com/actions/language-and-framework-guides/building-and-testing-java-with-gradle
+
+name: Test Runner PR Integration Tests
+
+on:
+  push:
+    branches: [ dev ]
+  pull_request:
+    branches: [ '**' ]
+
+jobs:
+  test-runner-integration:
+
+    runs-on: ubuntu-latest
+
+    services:
+      postgres:
+        image: postgres:12.3
+        env:
+          POSTGRES_PASSWORD: postgres
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+        ports:
+        - 5432:5432
+
+    steps:
+    - uses: actions/checkout@v2
+    # This step is needed when running test using a local server.
+    - name: Initialize Postgres DB
+      env:
+        PGPASSWORD: postgres
+      run: psql -h 127.0.0.1 -U postgres -f ./local-dev/local-postgres-init.sql
+    - name: Set up AdoptOpenJDK 11
+      uses: joschi/setup-jdk@v2
+      with:
+        java-version: 11
+    # This step caches Gradle packages for reuse in the same VM instance.
```

Comment:
Can you confirm this won't use stale artifacts? I.e. it knows how to update them between runs?

Is it an issue if a manual run caches newer artifacts and an hourly run picks those up instead of the right ones for the current build?

Reply:
this is only caching dependencies and gradle packages, and it uses the hash of our build file as (part of) the cache key, so any changes in dependencies will result in ignoring the cache.
(Branch name is also part of the cache key, so the nightly run on dev won't pick up a PR run from a separate branch)

Reply:
Thanks!